### ISSUE TITLE
Tutorial for re-training SynthSeg on non-brain datasets

### ISSUE THREAD

**Issue by naga-karthik** [BBillot/SynthSeg#87]
Hello! Thank you very much for the easy-to-follow tutorials on re-training SynthSeg! I am particularly interested in "Section 5.4: Extension to cardiac segmentation" of the paper. I was curious to know if there is a tutorial on how you were able to apply the model on cardiac images i.e. the specifcs of the intensity-based clustering using the EM algorithm? 

Ultimately, I wanted to test SynthSeg on the images of the spinal cord (SC). Given the requirement of fully-labeled scans for training SynthSeg, do you think it would be feasible to test this idea? Given an example of the SC (see below), this would mean that I would need to have the segmentations of the cord, cerebrospinal fluid, vertebrae, discs, the brain part at the top, etc. As SC datasets do not typically involve the segmentations of the structures other than the cord, I was hoping to use the intensity-based clustering approach that you described. Do you think that clustering intensities would work for so many structures in the image? Any thoughts would be highly appreciated, thank you! 

<details>
<summary> annotated cervical spinal cord MRI </summary>

<img width="663" alt="Screenshot 2024-06-26 at 3 45 04 PM" src="https://github.com/BBillot/SynthSeg/assets/53445351/bf780d4d-fa42-441f-84f4-06c6be608f9a">

</details>

**Comment by BBillot** [comment]
Hi and thanks for the interest in our work, it's always nice to see it inspires other people for their own research !

I think using the clustering approach to get dense segmentations for the unlabeled regions in your training data is a very valid approach. I would advise to use a different number of clusters for the background to capture different levels of granularity. You might also want to use the clustering technique for the foreground labels, especially those for which a single Gaussian doesn't accurately represent the intensity distribution.

I am not planning to release the clustering algorithm, simply because the code is very messy. But I'm sure there are plenty of codes/github repos out there for clustering (not necessarily with EM).

Good luck,
Benjamin
